File: src/runtime/js/std/sys_native.js
// Generated std/sys.js delegates host bindings through this native seam.

function _replaceArray(target, values) {
    const next = Array.isArray(values) ? Array.from(values, (value) => String(value)) : [];
    target.splice(0, target.length, ...next);
}

export const sys = {
    argv: Array.from(process.argv, (value) => String(value)),
    path: [],
    stderr: process.stderr,
    stdout: process.stdout,
    exit(code = 0) {
        process.exit(Number(code) || 0);
    },
};

export const argv = sys.argv;
export const path = sys.path;
export const stderr = sys.stderr;
export const stdout = sys.stdout;

export function exit(code) {
    return sys.exit(code);
}

export function set_argv(values) {
    _replaceArray(sys.argv, values);
}

export function set_path(values) {
    _replaceArray(sys.path, values);
}

export function write_stderr(text) {
    sys.stderr.write(String(text));
}

export function write_stdout(text) {
    sys.stdout.write(String(text));
}

sys.set_argv = set_argv;
sys.set_path = set_path;
sys.write_stderr = write_stderr;
sys.write_stdout = write_stdout;

File: src/runtime/cs/std/sys_native.cs
using System;
using System.Collections.Generic;

namespace Pytra.CsModule
{
    // Generated std/sys.cs delegates host bindings through this native seam.
    public static class sys_native
    {
        private static readonly List<string> _argv = new List<string>();
        private static readonly List<string> _path = new List<string>();

        public static List<string> argv { get { return _argv; } }
        public static List<string> path { get { return _path; } }

        public static void exit(long code)
        {
            System.Environment.Exit((int)code);
        }

        public static void set_argv(List<string> values)
        {
            _argv.Clear();
            _argv.AddRange(values);
        }

        public static void set_path(List<string> values)
        {
            _path.Clear();
            _path.AddRange(values);
        }

        public static void write_stderr(string text)
        {
            Console.Error.Write(text);
        }

        public static void write_stdout(string text)
        {
            Console.Write(text);
        }
    }
}
